xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/DeletePageOutcomePage.java

```diff
@@ -125,5 +125,22 @@ public DeletePageOutcomePage clickDeleteTerminalPage()
             elements.get(0).click();
         }
         return new DeletePageOutcomePage();
-    }   
+    }
+
+    /**
+     * Click on the link to view the deleted page.
+     *
+     * @param row the revision row for which to view the deleted page, starting at 1
+     * @return the view page of the deleted page
+     * @since 13.10.4
+     * @since 14.2RC1
+     */
+    public ViewPage clickViewDocument(int row)
+    {
+        WebElement link = getDriver()
+            .findElementWithoutWaiting(
+                By.cssSelector(String.format("table tbody tr:nth-child(%d) > td:nth-child(2) > a", row)));
```

Comment:
Not a big fan of such selector... Easy to break if we ever change the UI of the deleted pages. Isn't it possible to introduce an actual CSS class to semantically qualify such link?